src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/ArgumentSuggestions.cs

```diff
@@ -54,6 +55,12 @@ public static IEnumerable<KeyValuePair<string, DType>> GetArgumentSuggestions(In
                 return suggestor(tryGetEnumSymbol, suggestUnqualifiedEnums, scopeType, argumentIndex, out requiresSuggestionEscaping);
             }
 
+            // !!! This block will not be necessary once we move Join to Core
+            if (function is JoinFunction joinFunction)
+            {
+                return JoinSuggestion(tryGetEnumSymbol, suggestUnqualifiedEnums, scopeType, argumentIndex, out requiresSuggestionEscaping);
+            }
```

Comment:
@MikeStall

Reply:
Consider adding a new virtual to TexlFunction

virtual Type DeclarationType => this.GetType();

And then joinImpl can override:
  override Type DeclarationType => typeof(JoinFunction);